packages/bus/src/brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,479 @@
+    handlerInstance: EventHandler<T, R>,
+    eventName: string
+  ) {
+    const eventHandlers = this.handlers.get(eventName);
+    if (!eventHandlers) {
+      return;
+    }
+
+    const handlerIndex = eventHandlers.findIndex(
+      e =>
+        Object.getPrototypeOf(e).constructor.name ===
+        Object.getPrototypeOf(handlerInstance).constructor.name
+    );
+    if (handlerIndex !== -1) {
+      eventHandlers.splice(handlerIndex, 1);
+    }
+
+    if (eventHandlers.length === 0) {
+      this.handlers.delete(eventName);
+      await this.unbindQueue(eventName);
+    }
+  }
+
+  private async unbindQueue(eventName: string) {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    await this.channel.unbindQueue(
+      this.busConfig.clientQueue,
+      this.busConfig.exchange,
+      eventName
+    );
+  }
+
+  private sendCommandToQueue<T, R>(command: Command<T, R>) {
+    if (!this.channel) {
+      throw new IllegalOperation(this);
+    }
+
+    const { type, payload, correlationId, createdAt } = command;
+
+    this.channel.sendToQueue(
+      this.APP_QUEUE_NAME,
+      Buffer.from(JSON.stringify(payload)),
+      {
+        type,
+        correlationId,
+        contentType: 'application/json',
+        mandatory: true,
+        persistent: true,
+        timestamp: createdAt.getTime(),
+        replyTo: this.REPLY_QUEUE_NAME
+      }
+    );
+  }
+
+  private getHandlerInstance<T, R>(
+    type: EventHandlerConstructor<T, R>
+  ): EventHandler<T, R> {
+    if (!this.container.isRegistered(type)) {
+      throw new EventHandlerNotFound(type.name);
+    }
+
+    return this.container.resolve(type);
+  }
+
+  private expectReply<R>(correlationId: string, ttl: number): Promise<R> {
+    return Promise.race([
+      once(this.subject, correlationId)[0] as Promise<R>,
+      new Promise<never>((_, reject) =>
+        setTimeout(reject, ttl, new NoResponse(ttl)).unref()
+      )
+    ]);
+  }
+
+  private async reconnect(): Promise<void> {
+    try {
+      this.clear();
+      await this.init();
+    } catch (err) {
+      // add log
+    }
+  }
+
+  private reflectEventsNames(
+    handlerType: EventHandlerConstructor<unknown, unknown>,
+    handler: EventHandler<unknown, unknown>
+  ): string[] {
+    const types: (new (...args: unknown[]) => Event<unknown>)[] =
+      Reflect.getMetadata(Event, handlerType) ?? [];
+
+    if (!types.length) {
+      throw new NoSubscriptionsFound(handler);
+    }
+
+    return types.map(
+      (event: new (...args: unknown[]) => Event<unknown> | string) =>
+        typeof event === 'string' ? event : event.name
+    );
+  }
+
+  private async createConsumerChannel(): Promise<void> {
+    if (this.channel || !this.client) {
+      throw new IllegalOperation(this);
+    }
+
+    this.channel = await this.client.createConfirmChannel();
+
+    await this.bindExchangesToQueue();
+    await this.startBasicConsume();
+    await this.startReplyQueueConsume();
```

Comment:
Well, I don't know why you decided to raise an exception here.
```suggestion
    if (!this.channel && this.client) {
          this.channel = await this.client.createConfirmChannel();

          await this.bindExchangesToQueue();
          await this.startBasicConsume();
          await this.startReplyQueueConsume();
    }
```